### PR TITLE
First docs for porting

### PR DESCRIPTION
**Pull request by shartte**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -26,6 +26,36 @@ Contributing to NeoForge
...
+6. Run tests (see above)
+7. Fix any problems found by tests
+8. Create a squashed branch (i.e. `<mc_version>-squashed`)
+9. Generate patches (`unpackSourcePatches`)
+10. Apply formatting (`applyAllFormatting`)
+11. Remove MC sources and commit
+12. Squash all commits from "Initial base" created by the action, ensure the resulting commit contains no bot authorship (due to CLA signing issues later).
+13. **Make absolutly sure no Minecraft sources are still present in the history of the branch**
+14. If this is the first snapshot for a new version, publish this kits branch to a `port/<mc_version>` branch on the main repository, otherwise push just the new, squashed commit
+
```
Comment (Minecraftschurli) [inline docs/CONTRIBUTING.md]: idk if the contributing file is the right place for this since it is a maintainer exclusive thing to do

Comment (shartte) [reply docs/CONTRIBUTING.md]: Split into PORTING.md

File: docs/PORTING.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Neoform
+
+[NeoForm](https://github.com/neoforged/NeoForm) must be updated before anything can be done in NeoForge itself.
+
+## Neoforge
+
+1. Start Kits branch with [action](https://github.com/neoforged/actions/actions/workflows/start-kits.yml)
+   1. Press "Run workflow"
+   2. Fill out the survey
```
Comment (Minecraftschurli) [inline docs/PORTING.md]: survey -> fields

File: docs/PORTING.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      1. If it fails (i.e. with `Failed to download game artifact EXECUTABLE for CLIENT`), retry the job
+2. Clone the Kits repository and check out the branch corresponding to the Minecraft version
+3. Setup has already been run and `projects/neoforge` will contain the patched sources
+4. Fix rejected patch hunks found in the `rejects/` folder by re-applying the broken hunk and deleting the reject file
+5. Fix other compile errors
+6. Run tests (see above)
+7. Fix any problems found by tests
+8. Create a squashed branch (i.e. `<mc_version>-squashed`)
+9. Generate patches (`unpackSourcePatches`)
+10. Apply formatting (`applyAllFormatting`)
```
Comment (Minecraftschurli) [inline docs/PORTING.md]: create squash branch after generating the patches

File: docs/PORTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+11. Remove MC sources and commit
+12. Squash all commits from "Initial base" created by the action, ensure the resulting commit contains no bot authorship (due to CLA signing issues later).
```
Comment (Minecraftschurli) [inline docs/PORTING.md]: bot authorship is irrelevant now

**Review by Minecraftschurli — CHANGES_REQUESTED** [review]

File: docs/PORTING.md (hunk 4 above)
Comment (Matyrobbrt) [reply docs/PORTING.md]: It is still useless noise, the bot has no reason to be in the commits.

Comment (shartte) [reply docs/PORTING.md]: yeah, I'd like to keep this.

File: docs/PORTING.md (hunk 3 above)
Comment (shartte) [reply docs/PORTING.md]: Addressed

File: docs/PORTING.md (hunk 2 above)
Comment (shartte) [reply docs/PORTING.md]: Addressed

**Review by Minecraftschurli — APPROVED** [review]

**Pull request merged**